R/03-annotate.R

```diff
@@ -17,46 +19,196 @@
-
-  # This file is from https://depmap.org/portal/download/all/ and from DepMap Public 19Q3 All Files
-  # Essential gene labeling is from inst/extdata/Achilles_common_essentials.csv
-  # ctrl vs gene labels are from inst/extdata/pgPEN_annotations.txt
-
-  # This is the core of the code we'll need but we'll need to refactor
-  #d.annot <- d.annot %>%
-  #mutate(norm_ctrl_flag = case_when(
-  #  target_type == "gene_gene" ~ "double_targeting",
-  #  target_type == "gene_ctrl" & gene1_essential_flag == TRUE ~ "positive_control",
-  #  target_type == "ctrl_gene" & gene2_essential_flag == TRUE ~ "positive_control",
-  #  target_type == "gene_ctrl" & gene1_essential_flag != TRUE ~ "single_targeting",
-  #  target_type == "ctrl_gene" & gene2_essential_flag != TRUE ~ "single_targeting",
-  #  target_type == "ctrl_ctrl" ~ "negative_control")) %>%
-  #mutate(norm_ctrl_flag = factor(norm_ctrl_flag, levels = c("negative_control",
-  #                                                          "positive_control",
-  #                                                          "single_targeting",
-  #                                                          "double_targeting")))
-
-  gimap_dataset$annotation <- NULL #TODO: Final step is annotations that line up to the same order as the pg gene data should be stored here.
+  ############################ CONTROL GENE ANNOTATION #########################
+  # If control genes aren't provided then we get some from DepMap
+  if (!is.null(control_genes)) {
+    if (!file.exists(control_genes)) stop("The annotation_file specified cannot be found. Please double check the file path")
+    control_genes <- read_table(control_genes)[, 1]
+  } else {
+    # This file is from https://depmap.org/portal/download/all/ and from DepMap Public 19Q3 All Files
+    # Essential gene labeling is from inst/extdata/Achilles_common_essentials.csv
+    control_genes <- readr::read_tsv("https://figshare.com/ndownloader/files/40448429", show_col_types = FALSE)
+    control_genes <- control_genes %>%
+      tidyr::separate(col = Gene, into = c("gene_symbol", "entrez_id"), remove = FALSE, extra = "drop") %>%
+      dplyr::pull(gene_symbol)
+  }
+
+  ############################ Get TPM data ####################################
+  # This is not optional because its used to flag things
+  ## get TPM and CN information (w/ option for user to upload their own info)
+  depmap_metadata <- readr::read_csv("https://figshare.com/ndownloader/files/35020903", show_col_types = FALSE)
+
+  my_depmap_id <- depmap_metadata %>%
+    dplyr::filter(stripped_cell_line_name == cell_line) %>%
+    dplyr::pull(DepMap_ID)
+
+  tpm_file <- file.path(system.file("extdata", package = "gimap"), "CCLE_expression.csv")
+
+  if (!file.exists(tpm_file)) tpm_setup()
+
+  depmap_tpm <- readr::read_csv(tpm_file,
+                                show_col_types = FALSE,
+                                col_select = c("genes", dplyr::all_of(my_depmap_id))
```

Comment:
Since `tpm_setup()` reads in and cleans the data, if it's called, does this next chunk need to be called if `tpm_setup()` was run (because the `tpm_file` didn't exist)? I know that the `tpm_setup()` function isn't returning the cleaned df, and instead looks like it's returning the file path, but wondering if it's possible to simplify this I/O slightly. In other words, can we have `tpm_setup()` populate `depmap_tpm` and then only run the `read_csv()` chunk if `tpm_setup()` isn't run (e.g., file exists)?

Reply:
I guess the rename and mutate steps in the next lines after this would have to get moved to the setup function though?

Reply:
I think the confusion here is that only `tpm_setup()` is called if the file doesn't exist. Notice the lack of `{}` so lines 70 - 72 are ran regardless of the file's existence.

Reply:
Yes, that's how I understood it. My confusion/intent was to simplify I/O by only running `read_csv` if `tpm_setup()` didn't get called.

Reply:
`tpm_setup()` just downloads the file to the package folder and does some initial cleaning of the file but this only needs to be done once. But we always need to get this data into the environment by reading in the CSV. So read_csv() needs to be done regardless if the setup happened when the tpm_setup() was called just now or previously. 

Let me know if this makes sense and if so how I can make this clearer.